### PR TITLE
Added support for redirecting test output

### PR DESCRIPTION
**Pull request by oxtoacart**

### CODE REVIEW

File: testlog/testlog_test.go [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+	"bytes"
+	"testing"
+
+	"github.com/getlantern/golog"
+	"github.com/stretchr/testify/assert"
+)
+
+const (
+	expectedCapture = `ERROR mytest: testlog_test.go:24 error 1
+DEBUG mytest: buffer.go:54 debug 1
```
Comment (oxtoacart) [inline testlog/testlog_test.go:13]: This is a weird effect of writing to a buffer, not sure why it's happening.

File: golog.go [hunk 2]
```diff
@@ -87,6 +87,15 @@ func SetOutputs(errorOut io.Writer, debugOut io.Writer) {
 	})
 }
 
+// RedirectOutputs temporarily redirects outputs. Returns a function that resets outputs to their prior settings.
+func RedirectOutputs(errorOut io.Writer, debugOut io.Writer) (reset func()) {
```
Comment (joesis) [inline golog.go:91]: Alternatively we could modify `SetOutputs` to return the reset function, which has no effect on existing code if they don't use the return value. But this is also LGTM

File: testlog/testlog_test.go (hunk 1 above)
Comment (joesis) [reply testlog/testlog_test.go:13]: Yeah I noticed it somewhere as well. I bet it relates to how we call `runtime.Callers` and/or `runtime.FuncForPC` - just got the wrong pointer on the call stack.

**Review by joesis — APPROVED** [review]

File: golog.go (hunk 2 above)
Comment (oxtoacart) [reply golog.go:91]: Good idea, I'll do that!

**Comment by oxtoacart** [comment]
Thanks @joesis!

**Pull request merged**